clockwork_test.go

```diff
@@ -199,3 +201,33 @@ func TestFakeClockRace(t *testing.T) {
 	go func() { fc.NewTimer(d) }()
 	go func() { fc.Sleep(d) }()
 }
+
+func TestFakeClockSynchronousAfterFunc(t *testing.T) {
+	t.Parallel()
+	fc := NewFakeClock(WithSynchronousAfterFunc(true))
+	var called atomic.Bool
```

Comment:
Let's avoid atomic unless as a last resort. Here you can pretty readily use a channel:

`var called atomic.Bool` - > `called := make(chan bool)`

`called.Store(true)` -> `close(called)`

`if called.Load() { t.Falal(...) }` ->
```
select {
case <-called:
  t.Falal(...)
default:
}
```

Reply:
I actually realized that since this is called synchronously, there's no need for further synchronization, so I just used a bool. I ran the same test with the `-race` flag on, and I verified that the test passed; and then I also verified that the race detector test fails without the `WithSynchronousAfterFunc` option.